libeos-updater-util/flatpak.c

```diff
@@ -1457,6 +1457,651 @@ euu_filter_for_existing_flatpak_ref_actions (GHashTable *ref_actions  /* (elemen
+                                                               installed_related_refs,
+                                                               cancellable,
+                                                               error))
+        return FALSE;
+
+      /* If the source ref action is to uninstall then it should be
+       * go before its dependencies. */
+      if (ref_action->type == EUU_FLATPAK_REMOTE_REF_ACTION_UNINSTALL)
+        g_ptr_array_add (dependency_ref_actions,
+                         euu_flatpak_remote_ref_action_ref (ref_action));
+
+      /* Go through each of the related refs and add it to the dependency
+       * ref actions depending on what we are doing with the dependency
+       * and the state of the dependnecy */
+      for (j = 0; j < remote_related_refs->len; ++j)
+        {
+          EuuFlatpakRelatedRefsForRemote *related_refs_for_remote = g_ptr_array_index (remote_related_refs, j);
+          GHashTableIter iter;
+          gpointer key, value;
+
+          g_hash_table_iter_init (&iter, related_refs_for_remote->related_refs);
+
+          while (g_hash_table_iter_next (&iter, &key, &value))
+            {
+              FlatpakRelatedRef *related_ref = FLATPAK_RELATED_REF (key);
+              EuuFlatpakRemoteRefActionType action_type;
+
+              /* Is there something to do with this related ref? */
+              if (determine_action_for_related_ref (related_ref,
+                                                    ref_action,
+                                                    installed_related_refs,
+                                                    &action_type))
+                {
+                  FlatpakRemote *remote = related_refs_for_remote->remote;
+                  EuuFlatpakLocationRef *location_ref =
+                    euu_flatpak_location_ref_new (FLATPAK_REF (related_ref),
+                                                  flatpak_remote_get_name (remote),
+                                                  flatpak_remote_get_collection_id (remote));
+
+                  /* Dependencies have -1 as their serial number. At the point
+                   * at which dependencies are added, action ordering and prioritization
+                   * has already occurred, so the serial doesn't have much meaning. Set
+                   * it to -1 so that the flatpak installer knows not to update the counter. */
+                  g_ptr_array_add (dependency_ref_actions,
+                                   euu_flatpak_remote_ref_action_new (action_type,
+                                                                      location_ref,
+                                                                      ref_action->source,
+                                                                      -1));
+                }
+            }
+        }
+
+      /* If the source ref action is to install or update then it should be
+        * go before its dependencies. */
```

Comment:
‘should be go’

Reply:
s/before/after/

Reply:
Don’t you need to split it so that (if installing, for example), runtimes get installed first, then the app, then the app’s extension runtimes? Or do we not care about installing platform runtimes like `org.freedesktop.Platform`?

Reply:
As far as I can tell the flatpak API does not care if extension runtimes are installed completely separately before or after an app. I figured it'd be easier to handle both in the same way (eg, install all the related refs before the app).

Reply:
Probably worth putting that in a comment.